File: src/Complex/Prog10/Transverse/CFrameGraph.java
import java.awt.Color;
import java.awt.Graphics;

class CFrameGraph extends CFrame {
   public static final int kLCode = 1;
   public static final int kD1Code = 2;
   public static final int kD2Code = 3;
   public static final int kkCode = 4;
   public static final int kalphaCode = 5;
   public static final double kStartL = 1.2;
   public static final double kStartD1 = 0.15;
   public static final double kStartD2 = 0.1;
   public static final double kStartk = 1380000.0;
   public static final double kStartalpha = 0.4;
   public static final int kXo = 40;
   public static final int kYo = 370;
   public static final double kE = 2.0E11;
   public static final double kDensity = 7800.0;
   public static final double kScale = 1.0E8;
   public static final double kWmax = 200.0;
   public static final int kResponsePoints = 401;
   double[] mRec1 = new double[401];
   double[] mRec2 = new double[401];
   double[] mRec3 = new double[401];
   double mL = 1.2;
   double mD1 = 0.15;
   double mD2 = 0.1;
   double mk = 1380000.0;
   double malpha = 0.4;
   double mlambda1;
   double mlambda2;
   double mw;
   double mArea1;
   double mInertia1;
   double mArea2;
   double mInertia2;
   double mTime;

   public CFrameGraph(CFramePanel var1, int var2, int var3, int var4, int var5) {
      super(var1, var2, var3, var4, var5);
   }

   public void Frame(Graphics var1) {
      Color var5 = new Color(0.18F, 0.58F, 0.58F);
      new Color(0.77F, 0.38F, 0.0F);
      var1.setPaintMode();
      var1.setColor(Color.black);
      var1.drawLine(38, 370, 440, 370);
      var1.drawLine(38, 371, 440, 371);
      var1.drawLine(39, 374, 39, 15);
      var1.drawLine(40, 374, 40, 15);

      for (int var2 = 0; var2 < 8; var2++) {
         var1.drawLine(40, 370 - var2 * 50, 37, 370 - var2 * 50);
      }

      for (int var7 = 1; var7 < 5; var7++) {
         var1.drawLine(40 + var7 * 100, 370, 40 + var7 * 100, 374);
      }

      this.response();
      var1.setColor(Color.red);

      for (int var8 = 2; var8 < 400; var8++) {
         var1.drawLine(40 + (var8 - 1), (int)(370.0 - Math.abs(this.mRec1[var8 - 1] * 1.0E8)), 40 + var8, (int)(370.0 - Math.abs(this.mRec1[var8] * 1.0E8)));
      }

      var1.setColor(Color.blue);

      for (int var9 = 2; var9 < 400; var9++) {
         int var3 = (int)(370.0 - Math.abs(this.mRec2[var9 - 1] * 1.0E8));
         int var4 = (int)(370.0 - Math.abs(this.mRec2[var9] * 1.0E8));
         if (var3 > 0 & var4 > 0) {
            var1.drawLine(40 + (var9 - 1), var3, 40 + var9, var4);
         }
      }

      var1.setColor(var5);

      for (int var10 = 2; var10 < 400; var10++) {
         var1.drawLine(40 + (var10 - 1), (int)(370.0 - this.mRec3[var10 - 1] * 1.0E8), 40 + var10, (int)(370.0 - this.mRec3[var10] * 1.0E8));
      }

      var1.setColor(Color.white);
      var1.fillRect(0, 0, 440, 20);
   }

   public double log10(double var1) {
      return Math.log(var1) / Math.log(10.0);
   }

   public String nns(double var1, int var3) {
      if (var3 <= 0) {
         var3 = 1;
      }

      if (var1 == 0.0) {
         return new String("0");
      } else if (var1 < 0.0) {
         return new String("-" + this.nns(-var1, var3));
      } else {
         double var4 = Math.floor(this.log10(var1));
         double var6 = Math.pow(10.0, var4 - var3 + 1.0);
         long var8 = Math.round(var1 / var6);
         String var10 = new String(String.valueOf(var8 * var6));

         while (var10.length() > 1 && var10.indexOf(46) > -1) {
            boolean var11 = false;

            int var12;
            for (var12 = var10.length() - 1; var10.charAt(var12) == '0'; var11 = true) {
               var12--;
            }

            if (var11) {
               var10 = new String(var10.substring(0, var12 + 1));
            }

            String var13 = new String(var10.substring(0, var12));

            Double var14;
            try {
               var14 = new Double(var13);
            } catch (NumberFormatException var15) {
               break;
            }

            if (Math.abs(var1 - var14) > var6) {
               break;
            }

            var10 = var13;
         }

         return var10;
      }
   }

   public String nns(double var1) {
      return this.nns(var1, 4);
   }

   public void ControlMessage(CFrame var1, int var2, double var3) {
      boolean var5 = false;
      switch (var2) {
         case 1:
            this.mL = var3;
            this.response();
            break;
         case 2:
            this.mD1 = var3;
            this.response();
            break;
         case 3:
            this.mD2 = var3;
            this.response();
            break;
         case 4:
            this.mk = var3;
            this.response();
            break;
         case 5:
            this.malpha = var3;
            this.response();
      }

      if (var5) {
         this.repaint();
      }
   }

   public void response() {
      this.mArea1 = Math.PI * this.mD1 * this.mD1 / 4.0;
      this.mInertia1 = this.mArea1 * this.mD1 * this.mD1 / 16.0;
      this.mArea2 = Math.PI * this.mD2 * this.mD2 / 4.0;
      this.mInertia2 = this.mArea2 * this.mD2 * this.mD2 / 16.0;

      for (int var1 = 1; var1 < 401; var1++) {
         this.mw = var1 * 200.0 * 2.0 * Math.PI / 400.0;
         this.mlambda1 = Math.sqrt(Math.sqrt(this.mw * this.mw * this.mArea1 * 7800.0 / (2.0E11 * this.mInertia1)));
         this.mlambda2 = Math.sqrt(Math.sqrt(this.mw * this.mw * this.mArea2 * 7800.0 / (2.0E11 * this.mInertia2)));
         double var14 = this.mlambda1 * this.mL;
         double var16 = this.mlambda2 * this.mL;
         double var2 = Math.sin(var14);
         double var4 = Math.cos(var14);
         double var18 = Math.exp(var14);
         double var6 = (var18 - 1.0 / var18) / 2.0;
         double var8 = (var18 + 1.0 / var18) / 2.0;
         double var10 = var4 * var8 + 1.0;
         double var12 = var4 * var6 - var2 * var8;
         double var20 = 1.0 / (2.0E11 * this.mInertia1 * this.mlambda1 * this.mlambda1 * this.mlambda1 * var10);
         this.mRec1[var1] = -var12 * var20;
         var2 = Math.sin(var16);
         var4 = Math.cos(var16);
         var18 = Math.exp(var16);
         var6 = (var18 - 1.0 / var18) / 2.0;
         var8 = (var18 + 1.0 / var18) / 2.0;
         var10 = var4 * var8 + 1.0;
         var12 = var4 * var6 - var2 * var8;
         var20 = 1.0 / (2.0E11 * this.mInertia2 * this.mlambda2 * this.mlambda2 * this.mlambda2 * var10);
         this.mRec2[var1] = -var12 * var20;
         double var22 = this.mRec2[var1] + 1.0 / this.mk / (1.0 + this.malpha * this.malpha);
         double var24 = -this.malpha / this.mk / (1.0 + this.malpha * this.malpha);
         double var26 = 1.0 / this.mRec1[var1] + var22 / (var22 * var22 + var24 * var24);
         double var28 = -var24 / (var22 * var22 + var24 * var24);
         this.mRec3[var1] = 1.0 / Math.sqrt(var26 * var26 + var28 * var28);
      }
   }

   public double GetTime() {
      return this.mTime;
   }

   public boolean MouseEvent(int var1, boolean var2) {
      if (var2) {
         return false;
      } else {
         boolean var3 = this.inside(super.mFramePanel.mThisPt.x, super.mFramePanel.mThisPt.y);
         switch (var1) {
            case 0:
            case 1:
               double var4 = (super.mFramePanel.mThisPt.x - 40) * 200.0 / 400.0;
               if (var4 > 5000.0) {
                  var4 = 5000.0;
               }

               if (var4 < 50.0) {
                  var4 = 50.0;
               }
            default:
               return true;
         }
      }
   }
}

File: site/swingjs/j2s/Continuous/Prog12/Transverse/CFrameGraph.js
(function(){var P$=Clazz.newPackage("Continuous.Prog12.Transverse"),I$=[[0,'java.awt.Color','java.awt.Polygon']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "CFrameGraph", null, 'Continuous.Prog12.Transverse.CFrame');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
this.mRec=Clazz.array(Double.TYPE, [401]);
this.mxp=Clazz.array(Double.TYPE, [31]);
this.mAmp=Clazz.array(Double.TYPE, [31]);
this.mL=1.5;
this.mB=0.12;
this.mH=0.12;
this.mWanimate=500.0;
this.mTime=0.0;
},1);

C$.$fields$=[['D',['mL','mB','mH','mWanimate','mlambda','mw','mArea','mInertia','mTime'],'O',['mRec','double[]','+mxp','+mAmp']]]

Clazz.newMeth(C$, 'c$$Continuous_Prog12_Transverse_CFramePanel$I$I$I$I',  function (var1, var2, var3, var4, var5) {
;C$.superclazz.c$$Continuous_Prog12_Transverse_CFramePanel$I$I$I$I.apply(this,[var1, var2, var3, var4, var5]);C$.$init$.apply(this);
}, 1);

Clazz.newMeth(C$, 'Frame$java_awt_Graphics',  function (var1) {
var var3=Clazz.new_($I$(1,1).c$$F$F$F,[0.18, 0.58, 0.58]);
Clazz.new_($I$(1,1).c$$F$F$F,[0.77, 0.38, 0.0]);
var1.setPaintMode$();
var1.setColor$java_awt_Color($I$(1).black);
var1.drawLine$I$I$I$I(38, 370, 440, 370);
var1.drawLine$I$I$I$I(38, 371, 440, 371);
var1.drawLine$I$I$I$I(39, 374, 39, 15);
var1.drawLine$I$I$I$I(40, 374, 40, 15);
for (var var2=0; var2 < 12; var2++) {
var1.drawLine$I$I$I$I(40, 370 - var2 * 30, 37, 370 - var2 * 30);
}
for (var var5=1; var5 < 6; var5++) {
var1.drawLine$I$I$I$I(40 + var5 * 80, 370, 40 + var5 * 80, 367);
}
var1.setColor$java_awt_Color(var3);
this.response$();
for (var var6=2; var6 < 400; var6++) {
var1.drawLine$I$I$I$I(40 + (var6 - 1), ((10.0 - Math.log(Math.abs(this.mRec[var6 - 1])) / Math.log(10.0) * 30.0)|0), 40 + var6, ((10.0 - Math.log(Math.abs(this.mRec[var6])) / Math.log(10.0) * 30.0)|0));
}
var var7=((this.mWanimate * 400.0 / 5000.0)|0);
var1.setColor$java_awt_Color($I$(1).blue);
var1.drawLine$I$I$I$I(40 + var7, 370, 40 + var7, ((10.0 - Math.log(Math.abs(this.mRec[var7])) / Math.log(10.0) * 30.0)|0));
var1.drawLine$I$I$I$I(40 + var7, 373, 40 + var7, 368);
var1.drawString$S$I$I(this.nns$D(this.mWanimate) + " Hz", 40 + var7 + 2 , 385);
});

Clazz.newMeth(C$, 'log10$D',  function (var1) {
return Math.log(var1) / Math.log(10.0);
});

Clazz.newMeth(C$, 'nns$D$I',  function (var1, var3) {
if (var3 <= 0) {
var3=1;
}if (var1 == 0.0 ) {
return "0";
} else if (var1 < 0.0 ) {
return "-" + this.nns$D$I(-var1, var3);
} else {
var var4=Math.floor(this.log10$D(var1));
var var6=Math.pow(10.0, var4 - var3 + 1.0);
var var8=Math.round$D(var1 / var6);
var var10=String.valueOf$D(var8 * var6);
while (var10.length$() > 1 && var10.indexOf$I(46) > -1 ){
var var11=false;
var var12;
for (var12=var10.length$() - 1; var10.charAt$I(var12) == "0"; var11=true) {
--var12;
}
if (var11) {
var10=var10.substring$I$I(0, var12 + 1);
}var var13=var10.substring$I$I(0, var12);
var var14;
try {
var14=java.lang.Double.parseDouble$S(var13);
} catch (var15) {
if (Clazz.exceptionOf(var15,"NumberFormatException")){
break;
} else {
throw var15;
}
}
if (Math.abs(var1 - var14) > var6 ) {
break;
}var10=var13;
}
return var10;
}});

Clazz.newMeth(C$, 'nns$D',  function (var1) {
return this.nns$D$I(var1, 4);
});

Clazz.newMeth(C$, 'DrawModeShape$java_awt_Graphics$I$I',  function (var1, var2, var3) {
var1.setColor$java_awt_Color($I$(1).white);
var1.fillRect$I$I$I$I(0, 390, 600, 190);
var1.fillRect$I$I$I$I(0, 0, 450, 20);
var var8=20;
var var10=30;
var var11=1.0472;
var var13=($s$[0] = 470, $s$[0]);
var var15=Clazz.new_($I$(1,1).c$$F$F$F,[0.18, 0.58, 0.58]);
var var16=Clazz.new_($I$(1,1).c$$F$F$F,[0.52, 0.82, 0.82]);
var var17=Clazz.new_($I$(1,1).c$$F$F$F,[0.77, 0.38, 0.0]);
Clazz.new_($I$(1,1).c$$F$F$F,[1.0, 1.0, 0.4]);
var var20=90;
var var21=((var20 * Math.cos(var11) / 2.0)|0);
this.setOmegaDisp$D(this.mWanimate);
this.mAmp[0]=0.0;
for (var var4=0; var4 < 31; var4++) {
this.mAmp[var4]=this.mxp[var4] * Math.cos(0.7 * this.mTime * 2.0 * 3.141592653589793 );
}
var1.setColor$java_awt_Color(var16);
var var23=Clazz.new_($I$(2,1));
var var24=((250.0 * this.mB * Math.cos(var11)  / 2.0)|0);
var var25=((250.0 * this.mB * Math.sin(var11)  / 2.0)|0);
for (var var28=0; var28 < 31; var28++) {
var23.addPoint$I$I(var8 + var10 + var21 + ((var28 * 250.0 * this.mL  / 30.0)|0)  - var24, var13 - ((this.mAmp[var28] - this.mH * 250.0 / 2.0)|0) + var25);
}
var23.addPoint$I$I(var8 + var10 + var21 + ((250.0 * this.mL)|0)  - var24, var13 - ((this.mAmp[30] + this.mH * 250.0 / 2.0)|0) + var25);
for (var var29=29; var29 > -1; var29--) {
var23.addPoint$I$I(var8 + var10 + var21 + ((var29 * 250.0 * this.mL  / 30.0)|0)  - var24, var13 - ((this.mAmp[var29] + this.mH * 250.0 / 2.0)|0) + var25);
}
var1.drawPolygon$java_awt_Polygon(var23);
var1.fillPolygon$java_awt_Polygon(var23);
var1.setColor$java_awt_Color(var15);
var var26=Clazz.new_($I$(2,1));
for (var var30=0; var30 < 31; var30++) {
var26.addPoint$I$I(var8 + var10 + var21 + ((var30 * 250.0 * this.mL  / 30.0)|0)  - var24, var13 - ((this.mAmp[var30] + this.mH * 250.0 / 2.0)|0) + var25);
}
var26.addPoint$I$I(var8 + var10 + var21 + ((250.0 * this.mL)|0) + var24 , var13 - ((this.mAmp[30] + this.mH * 250.0 / 2.0)|0) - var25 );
for (var var31=29; var31 > -1; var31--) {
var26.addPoint$I$I(var8 + var10 + var21 + ((var31 * 250.0 * this.mL  / 30.0)|0) + var24 , var13 - ((this.mAmp[var31] + this.mH * 250.0 / 2.0)|0) - var25 );
}
var1.drawPolygon$java_awt_Polygon(var26);
var1.fillPolygon$java_awt_Polygon(var26);
var1.setColor$java_awt_Color(var15);
var var27=Clazz.new_($I$(2,1));
var27.addPoint$I$I(var8 + var10 + var21 + ((250.0 * this.mL)|0)  - var24, var13 - ((this.mAmp[30] + this.mH * 250.0 / 2.0)|0) + var25);
var27.addPoint$I$I(var8 + var10 + var21 + ((250.0 * this.mL)|0) + var24 , var13 - ((this.mAmp[30] + this.mH * 250.0 / 2.0)|0) - var25 );
var27.addPoint$I$I(var8 + var10 + var21 + ((250.0 * this.mL)|0) + var24 , var13 - ((this.mAmp[30] - this.mH * 250.0 / 2.0)|0) - var25 );
var27.addPoint$I$I(var8 + var10 + var21 + ((250.0 * this.mL)|0)  - var24, var13 - ((this.mAmp[30] - this.mH * 250.0 / 2.0)|0) + var25);
var1.drawPolygon$java_awt_Polygon(var27);
var1.fillPolygon$java_awt_Polygon(var27);
var1.setColor$java_awt_Color(var17);
var1.drawLine$I$I$I$I(var8 + var10 + ((250.0 * this.mL)|0) + var21 , var13 - (this.mAmp[30]|0), var8 + var10 + ((250.0 * this.mL)|0) + var21 + 20 , var13 - (this.mAmp[30]|0));
var var14=((20.0 * Math.cos(0.7 * this.mTime * 2.0 * 3.141592653589793 ))|0);
if (var14 != 0) {
if (var14 > 0) {
var1.drawLine$I$I$I$I(var8 + var10 + ((250.0 * this.mL)|0) + var21 + 12 , var13 - (this.mAmp[30]|0), var8 + var10 + ((250.0 * this.mL)|0) + var21 + 12 , var13 - var14 - (this.mAmp[30]|0) );
var1.drawLine$I$I$I$I(var8 + var10 + ((250.0 * this.mL)|0) + var21 + 13 , var13 - (this.mAmp[30]|0), var8 + var10 + ((250.0 * this.mL)|0) + var21 + 13 , var13 - var14 - (this.mAmp[30]|0) - 1 );
var1.drawLine$I$I$I$I(var8 + var10 + ((250.0 * this.mL)|0) + var21 + 14 , var13 - (this.mAmp[30]|0), var8 + var10 + ((250.0 * this.mL)|0) + var21 + 14 , var13 - var14 - (this.mAmp[30]|0) );
}if (var14 < 0) {
var1.drawLine$I$I$I$I(var8 + var10 + ((250.0 * this.mL)|0) + var21 + 12 , var13 - (this.mAmp[30]|0), var8 + var10 + ((250.0 * this.mL)|0) + var21 + 12 , var13 - var14 - (this.mAmp[30]|0) );
var1.drawLine$I$I$I$I(var8 + var10 + ((250.0 * this.mL)|0) + var21 + 13 , var13 - (this.mAmp[30]|0), var8 + var10 + ((250.0 * this.mL)|0) + var21 + 13 , var13 - var14 - (this.mAmp[30]|0)  + 1);
var1.drawLine$I$I$I$I(var8 + var10 + ((250.0 * this.mL)|0) + var21 + 14 , var13 - (this.mAmp[30]|0), var8 + var10 + ((250.0 * this.mL)|0) + var21 + 14 , var13 - var14 - (this.mAmp[30]|0) );
}}});

Clazz.newMeth(C$, 'ControlMessage$Continuous_Prog12_Transverse_CFrame$I$D',  function (var1, var2, var3) {
var var5=false;
switch (var2) {
case 1:
this.mL=var3;
this.response$();
break;
case 2:
this.mB=var3;
this.response$();
break;
case 3:
this.mH=var3;
this.response$();
break;
case 4:
this.mWanimate=var3;
this.setOmegaDisp$D(this.mWanimate);
var5=true;
break;
case 5:
this.mTime+=var3;
var5=true;
}
if (var5) {
this.repaint$();
}});

Clazz.newMeth(C$, 'response$',  function () {
this.mArea=3.141592653589793 * this.mB * this.mH ;
this.mInertia=this.mArea * this.mH * this.mH  / 12.0;
for (var var1=1; var1 < 401; var1++) {
this.mw=var1 * 5000.0 * 2.0 * 3.141592653589793  / 400.0;
this.mlambda=Math.sqrt(Math.sqrt(this.mw * this.mw * this.mArea * 7800.0  / (2.0E11 * this.mInertia)));
var var14=this.mlambda * this.mL;
var var2=Math.sin(var14);
var var4=Math.cos(var14);
var var16=Math.exp(var14);
var var6=(var16 - 1.0 / var16) / 2.0;
var var8=(var16 + 1.0 / var16) / 2.0;
var var10=var4 * var8 - 1.0;
var var12=var4 * var6 - var2 * var8;
var var18=1.0 / (2.0E11 * this.mInertia * this.mL * this.mL * this.mL * var10 );
this.mRec[var1]=-var12 * var18;
}
this.setOmegaDisp$D(this.mWanimate);
});

Clazz.newMeth(C$, 'setOmegaDisp$D',  function (var1) {
this.mArea=3.141592653589793 * this.mB * this.mH ;
this.mInertia=this.mArea * this.mH * this.mH  / 12.0;
this.mlambda=Math.sqrt(Math.sqrt(var1 * 2.0 * 3.141592653589793 * var1 * 2.0 * 3.141592653589793 * this.mArea * 7800.0  / (2.0E11 * this.mInertia)));
var var12=this.mlambda * this.mL;
var var14=Math.sin(var12);
var var16=Math.cos(var12);
var var22=Math.exp(var12);
var var18=(var22 - 1.0 / var22) / 2.0;
var var20=(var22 + 1.0 / var22) / 2.0;
var var6=var14 - var18;
var var8=var16 - var20;
var var4=var16 * var20 - 1.0;
var var10=1.0 / (2.0E11 * this.mInertia * this.mL * this.mL * this.mL * var4 );
var var34=0.0;
for (var var3=0; var3 < 31; var3++) {
var var24=this.mlambda * var3 * this.mL  / 30.0;
var var26=Math.sin(var24);
var var28=Math.cos(var24);
var22=Math.exp(var24);
var var30=(var22 - 1.0 / var22) / 2.0;
var var32=(var22 + 1.0 / var22) / 2.0;
this.mxp[var3]=(var6 * (var28 + var32) - var8 * (var26 + var30)) * var10;
if (Math.abs(this.mxp[var3]) > var34 ) {
var34=Math.abs(this.mxp[var3]);
}}
for (var var36=0; var36 < 31; var36++) {
this.mxp[var36]=this.mxp[var36] * 20.0 / var34;
}
});

Clazz.newMeth(C$, 'GetTime$',  function () {
return this.mTime;
});

Clazz.newMeth(C$, 'MouseEvent$I$Z',  function (code, prevh1t) {
if (prevh1t) {
return false;
} else {
switch (code) {
case 0:
case 1:
var newOmega=(this.mFramePanel.mThisPt.x - 40) * 5000.0 / 400.0;
if (newOmega > 5000.0 ) {
newOmega=5000.0;
}if (newOmega < 20.0 ) {
newOmega=20.0;
}this.ControlMessage$Continuous_Prog12_Transverse_CFrame$I$D(this, 4, newOmega);
default:
return true;
}
}});
var $s$ = new Int16Array(1);

Clazz.newMeth(C$);
})();
;Clazz.setTVer('5.0.1-v7');//Created 2025-10-14 19:43:49 Java2ScriptVisitor version 5.0.1-v7 net.sf.j2s.core.jar version 5.0.1-v7
